### PR TITLE
Fix cluster validation when there's additional nodes. 

### LINKED ISSUE #2379: Validation fails if ready nodes != Min nodes
```
houseparty k8s.kerkstraat.secrethouse.party ted:~$ kops validate cluster
Using cluster from kubectl context: k8s.kerkstraat.secrethouse.party

Validating cluster k8s.kerkstraat.secrethouse.party

INSTANCE GROUPS
NAME			ROLE	MACHINETYPE	MIN	MAX	SUBNETS
master-eu-central-1a	Master	m3.large	1	1	eu-central-1a
nodes			Node	t2.medium	3	5	eu-central-1a,eu-central-1b

NODE STATUS
NAME						ROLE	READY
ip-172-20-37-120.eu-central-1.compute.internal	master	True
ip-172-20-38-30.eu-central-1.compute.internal	node	True
ip-172-20-46-218.eu-central-1.compute.internal	node	True
ip-172-20-59-88.eu-central-1.compute.internal	node	True
ip-172-20-88-125.eu-central-1.compute.internal	node	True
ip-172-20-89-252.eu-central-1.compute.internal	node	True

Validation Failed
Ready Master(s) 1 out of 1.
Ready Node(s) 5 out of 3.

your nodes are NOT ready k8s.kerkstraat.secrethouse.party
houseparty k8s.kerkstraat.secrethouse.party ted:~$ kops edit ig nodes
Using cluster from kubectl context: k8s.kerkstraat.secrethouse.party
# Changed nodes min from 3 -> 5

^[[Ahouseparty k8s.kerkstraat.secrethouse.party ted:~$ kops validate cluster
Using cluster from kubectl context: k8s.kerkstraat.secrethouse.party

Validating cluster k8s.kerkstraat.secrethouse.party

INSTANCE GROUPS
NAME			ROLE	MACHINETYPE	MIN	MAX	SUBNETS
master-eu-central-1a	Master	m3.large	1	1	eu-central-1a
nodes			Node	t2.medium	5	5	eu-central-1a,eu-central-1b

NODE STATUS
NAME						ROLE	READY
ip-172-20-37-120.eu-central-1.compute.internal	master	True
ip-172-20-38-30.eu-central-1.compute.internal	node	True
ip-172-20-46-218.eu-central-1.compute.internal	node	True
ip-172-20-59-88.eu-central-1.compute.internal	node	True
ip-172-20-88-125.eu-central-1.compute.internal	node	True
ip-172-20-89-252.eu-central-1.compute.internal	node	True

Your cluster k8s.kerkstraat.secrethouse.party is ready
```
cc @justinsb , who helped me diagnose in slack.

### PR DESCRIPTION
**Pull request by GauntletWizard**
Fixes #2379 

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2380)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

File: pkg/validation/validate_cluster.go [hunk 1]
```diff
@@ -180,7 +180,7 @@ func validateTheNodes(clusterName string, validationCluster *ValidationCluster)
 	}
 
 	validationCluster.NodesReady = true
-	if len(validationCluster.NodesNotReadyArray) != 0 || validationCluster.NodesCount != len(validationCluster.NodesReadyArray) {
+	if len(validationCluster.NodesNotReadyArray) != 0 || validationCluster.NodesCount > len(validationCluster.NodesReadyArray) {
```
Comment (chrislovecnm) [inline pkg/validation/validate_cluster.go]: Should this be >= ?? It is late and I am on my cell... so silly question.

**Comment by GauntletWizard** [comment]
I initially wrote it that way, but this is a negative (if true, fail); if
it's equal then we're fine (we've got precisely the number of boxes we're
looking for (and don't want to hit the exception path)

On Apr 18, 2017 20:38, "Chris Love" <notifications@github.com> wrote:

*@chrislovecnm* commented on this pull request.

Question for u
------------------------------

In pkg/validation/validate_cluster.go
<https://github.com/kubernetes/kops/pull/2380#discussion_r112109838>:

> @@ -180,7 +180,7 @@ func validateTheNodes(clusterName string, validationCluster *ValidationCluster)
 	}

 	validationCluster.NodesReady = true
-	if len(validationCluster.NodesNotReadyArray) != 0 ||
validationCluster.NodesCount != len(validationCluster.NodesReadyArray)
{
+	if len(validationCluster.NodesNotReadyArray) != 0 ||
validationCluster.NodesCount > len(validationCluster.NodesReadyArray)
{

Should this be >= ?? It is late and I am on my cell... so silly question.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kops/pull/2380#pullrequestreview-33399517>,
or mute the thread

**Pull request merged**